src/lib/bootloader/grub2base.rb

```diff
@@ -45,13 +45,22 @@ class Grub2Base < BootloaderBase
 
     attr_accessor :pmbr_action
 
-    # @return [Boolean]
+    # @!attribute trusted_boot
+    #   @return [Boolean] current trusted boot setting
     attr_accessor :trusted_boot
 
+    # @!attribute secure_boot
+    #   @return [Boolean] current secure boot setting
+    attr_accessor :secure_boot
+
     # @!attribute console
     #   @return [::Bootloader::SerialConsole] serial console or nil if none
     attr_reader :console
 
+    # @!attribute stage1
+    #   @return [::Bootloader::Stage1, nil] bootloader stage1, if one is needed
+    attr_reader :stage1
```

Comment:
this is a bit strange as grub2EFI does not have stage1. So original intention is to have it only in grub2.rb and if something needs to depend on it, it should check bootloader or ask if method is defined.

Reply:
That's a valid point and I considered doing it that way. The usage is here:

https://github.com/yast/yast-bootloader/blob/master/src/lib/bootloader/proposal_client.rb#L338-L339

I my view this is a matter of class design. This proposal client code should work with any bootloader. So, basically with Grub2Base. That's why I decided to move the stage1 method.

Inconsistent class interfaces are imho one of the reasons yast code so easily runs into exceptions.